rw_core/main.go

```diff
@@ -98,22 +96,31 @@ func newRWCore(cf *config.RWCoreFlags) *rwCore {
 func (rw *rwCore) start(ctx context.Context, instanceID string) {
 	logger.Info("Starting RW Core components")
 
+	var err error
+	err = kvstore.ValidateAddress(rw.config.KVStoreAddress)
+	if err != nil {
+		logger.Fatal("Invalid address")
+	}
+
 	// Setup KV Client
 	logger.Debugw("create-kv-client", log.Fields{"kvstore": rw.config.KVStoreType})
-	var err error
 	if rw.kvClient, err = newKVClient(
 		rw.config.KVStoreType,
-		rw.config.KVStoreHost+":"+strconv.Itoa(rw.config.KVStorePort),
+		rw.config.KVStoreAddress,
 		rw.config.KVStoreTimeout); err != nil {
 		logger.Fatal(err)
 	}
-	cm := conf.NewConfigManager(rw.kvClient, rw.config.KVStoreType, rw.config.KVStoreHost, rw.config.KVStorePort, rw.config.KVStoreTimeout)
+	cm := conf.NewConfigManager(rw.kvClient, rw.config.KVStoreType, rw.config.KVStoreAddress, rw.config.KVStoreTimeout)
 	go conf.StartLogLevelConfigProcessing(cm, ctx)
 
+	err = kvstore.ValidateAddress(rw.config.KafkaAdapterAddress)
```

Comment:
I see that ValidateAddress is a generic method; then should be in common utils package and not kvstore.

Also, would suggest to introduce this check in config loading step itself as a new Value type https://golang.org/pkg/flag/#Value

Reply:
removed this code and added the required check in the config file "rw_core/config/config.go"